src/index.js

```diff
@@ -73,12 +78,85 @@ export function display() {
               <li class="item">
               ${item.description} 
               </li>
-              <i class="fas fa-ellipsis-v"></i>
+              <i id=${item.id}  class="fas fa-trash-alt"></i>
           </form>
           `;
       }
     });
     localStorage.setItem('todoLists', JSON.stringify(todoLists));
   }
 }
-display();
\ No newline at end of file
+display();
+
+const input = document.getElementById('input');
+const enter = document.getElementById('enter-key');
+function addLocalStorage() {
+  const localStore = JSON.stringify(todoLists);
+  localStorage.setItem('todoLists', localStore);
+  display();
+  window.location.reload(false);
+}
+if (localStorage.getItem('todoLists') !== null) {
+  todoLists = JSON.parse(localStorage.getItem('todoLists'));
+  display();
+}
+enter.addEventListener('click', () => {
+  todoLists.push({ description: input.value, completed: false, index: todoLists.length });
+  display();
+  addLocalStorage();
+});
+function remove(id) {
+  todoLists.splice(id, 1);
+  display();
+  addLocalStorage();
+}
+const btn = document.getElementsByClassName('fa-trash-alt');
+Array.from(btn).forEach((item, i) => {
+  item.addEventListener('click', () => {
+    remove(i);
+  });
+});
+
+function clear() {
+  todoLists = todoLists.filter((item) => item.completed === false);
+  display();
+  addLocalStorage();
+}
+
+const clean = document.getElementsByClassName('clear-completed');
+Array.from(clean).forEach((btn) => {
+  btn.addEventListener('click', () => {
+    clear();
+  });
+});
+// edit
+const area = document.getElementsByClassName('textarea');
+function edittodo(i) {
+  area[i].style.display = 'inline';
+  const line = document.querySelectorAll('.item');
+  line[i].style.display = 'none';
+  area[i].addEventListener('change', () => {
+    area[i].textContent = area[i].value;
+    todoLists[i].description = area[i].textContent;
+    addLocalStorage();
+  });
+}
+const editbtn = document.getElementsByTagName('button');
+Array.from(editbtn).forEach((edit, i) => {
+  edit.addEventListener('click', (e) => {
+    e.preventDefault();
+    editbtn[i].innerText = 'save';
+    edittodo(i - 1);
+  });
+});
+function indexLoop() {
+  if (todoLists.length > 0) {
+    let i = 0;
+    while (i < todoLists.length) {
+      todoLists[i].index = i;
```

Comment:
- [ ] New entries **do not match** the property index. The _property index_ must be set to the value of the _new array length_ (i.e. if a 5th task is added to the list, the index of that task should equal 5).

**Index mismatch in your application:**
![image](https://user-images.githubusercontent.com/12745474/140606680-e56c755d-be90-401c-a40b-a56b5adc02e9.png)

Reply:
Fixed, thank you.

Reply:
This issue has been addressed well. Nice job!